packages/expo-cli/src/commands/run/android/runAndroid.ts

```diff
@@ -101,9 +102,16 @@ async function resolveOptionsAsync(
     port = 8081;
   }
 
-  const variant = options.variant.toLowerCase();
-  const apkDirectory = Android.getAPKDirectory(projectRoot);
-  const apkVariantDirectory = path.join(apkDirectory, variant);
+  // TODO: why would this be different? Can we get the different name?
+  const appName = 'app';
+
+  const apkDirectory = path.join(projectRoot, 'android', appName, 'build', 'outputs', 'apk');
+
+  // buildDeveloperTrust -> build, developer, trust (where developer, and trust are flavors).
+  const [buildType, ...flavorDimensions] = options.variant
+    .split(/(?=[A-Z])/)
```

Comment:
could use some comment

Reply:
also, I'm not saying we should care about that case, but just FYI

if you have flavor e.g. `exaMPle` the gradle command would look like this `:app:buildExaMPleRelease`

Reply:
I think for the time being, the current implementation is a fair compromise.